File: src/main/java/com/leijendary/spring/boot/template/api/v1/data/SampleTranslationRequest.java
package com.leijendary.spring.boot.template.api.v1.data;

import javax.validation.constraints.NotBlank;

import com.leijendary.spring.boot.template.data.TranslationRequest;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Data
@EqualsAndHashCode(callSuper = true)
public class SampleTranslationRequest extends TranslationRequest {

    @NotBlank(message = "validation.required")
    private String name;

    @NotBlank(message = "validation.required")
    private String description;
}

File: src/main/kotlin/com/leijendary/spring/boot/template/api/v1/data/SampleTranslationRequest.kt
package com.leijendary.spring.boot.template.api.v1.data

import com.leijendary.spring.boot.template.core.data.TranslationRequest
import javax.validation.constraints.NotBlank

data class SampleTranslationRequest(
    @get:NotBlank(message = "validation.required")
    val name: String? = null,

    @get:NotBlank(message = "validation.required")
    val description: String? = null
) : TranslationRequest()